src/main/argv.js

```diff
@@ -0,0 +1,23 @@
+import minimist from 'minimist';
+
+// inspired by yargs' process-argv
+export const isElectronApp = () => !!process.versions.electron;
+export const isElectronBundledApp = () => isElectronApp() && !process.defaultApp;
+
+export const parseAndTrimArgs = argv => {
+    // bundled Electron app: ignore 1 from "my-app arg1 arg2"
+    // unbundled Electron app: ignore 2 from "electron main/index.js arg1 arg2"
+    // node.js app: ignore 2 from "node src/index.js arg1 arg2"
+    const ignoreCount = isElectronBundledApp() ? 1 : 2;
+
+    const parsed = minimist(argv);
+
+    // ignore arguments AFTER parsing to handle cases like "electron --inspect=42 my.js arg1 arg2"
```

Comment:
I don't totally understand this comment

Reply:
The call to `minimist` will pull out any "options" arguments -- the ones starting with `-` like `-f` or `--inspect=12345` -- and then put everything else into an array. The goal in this code is to ignore the first `ignoreCount` things which are NOT options, so it needs to do the `slice` on the `argv`-like array coming out of `minimist` (called `parsed._` here) rather than the `argv` array going into `minimist`.